### PR TITLE
WIP: Ignore leading whitespace for list types

### PR DESCRIPTION
**Pull request by harshad1**
This PR addresses [Markdown: Add support for nested check lists](https://github.com/gsantner/markor/issues/794).

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java [hunk 1]
```diff
@@ -107,14 +107,24 @@ protected void setBarVisible(ViewGroup barLayout, boolean visible) {
...
         }
 
+        if (ignoreLeadingWhitespace) {
+            for (; i < (cursor - 1); ++i) {
+                char c = text.charAt(i + 1);
+                if (c != ' ' && c != '\t') {
+                    break;
+                }
+            }
+        }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java]: Linear search forward from newline. Should not cause noticeable performance hit.

**Comment by harshad1** [comment]
@gsantner What would it take to get this reviewed and merged?

**Comment by gsantner** [comment]
How do you practically differenciate between if its whitespace for CodeBlock, OR list item?

**Comment by westurner** [comment]
I haven't reviewed the code in any depth (and haven't written Java in many years), but could you track the state of the previous line with e.g. `int currentListIndent`?

IIUC, codeblocks can be included in nested lists?

```markdown

currentListIndent=0

    # Code block
    import java.util

- 1 (currentListIndent=2)
  - A (currentListIndent=4)

        # Code block
        import java.util
```

But if you want to indicate the code language, it's better to use a fenced code block:

```markdown
- 1 (currentListIndent=2)
  - A (currentListIndent=4)

    ```java
    # Code block
    import java.util
    ````
```

**Comment by harshad1** [comment]
> How do you practically differenciate between if its whitespace for CodeBlock, OR list item?

I do not (and I do not know how to). My approach here is to simply ignore any whitespace between the (previous) newline and the first non-whitespace character. With this change, if you were to tap the list button (say) with the cursor on an indented code block, it would add a leading bullet at the same indent as the code block. To me this is the least-surprising behavior, but I am open to criticism and suggestions.

**Comment by gsantner** [comment]
Please just do some tries and verify it works for your usecase and that one from issue(s).

I will merge it when you tell me it's ready.

**Comment by harshad1** [comment]
> Please just do some tries and verify it works for your usecase and that one from issue(s).
> 
> I will merge it when you tell me it's ready.

Ok. I am doing some more testing and am trying to get the reporter from that issue to do some testing too. 

I will let you know when the PR is ready. This may take a day or two.

It may be useful if you create 'work and progress' and 'ready to merge' labels in github

**Comment by gsantner** [comment]
I guess most of reporting people dont know about code and stuff. Should be fine when you confirm.

**Pull request merged**